packages/prosemirror-resizable-view/src/prosemirror-resizable-view.ts

```diff
@@ -1,172 +1,278 @@
+      });
 
-      handle.dataset.dragging = 'true';
+      outer.addEventListener('mouseout', () => {
+        handleBottom.setHandleDisplay();
+        handleBottomRight.setHandleDisplay();
+        handleBottomLeft.setHandleDisplay();
+      });
 
-      const startX = e.pageX;
+      handleBottom.dom.addEventListener('mousedown', (e) =>
+        this.resizeHandler(e, view, getPos, handleBottom),
+      );
 
-      const startWidth = getWidthFromNode(this.node) || getSizeFromDom(inner)[0];
+      handleBottomRight.dom.addEventListener('mousedown', (e) =>
+        this.resizeHandler(e, view, getPos, handleBottomRight),
+      );
 
-      const onMouseMove = throttle(100, false, (e: MouseEvent) => {
-        const currentX = e.pageX;
+      handleBottomLeft.dom.addEventListener('mousedown', (e) =>
+        this.resizeHandler(e, view, getPos, handleBottomLeft),
+      );
+      outer.append(handleBottom.dom);
+      outer.append(handleBottomRight.dom);
+      outer.append(handleBottomLeft.dom);
+    }
 
-        const diffX = currentX - startX;
-        outer.style.width = `${startWidth + diffX}px`;
-      });
+    outer.append(element);
 
-      const onMouseUp = (e: MouseEvent) => {
-        e.preventDefault();
+    this.dom = outer;
+    this.#node = node;
+    this.#element = element;
+    this.aspectRatio = aspectRatio;
+  }
 
-        handle.dataset.dragging = '';
-        setHandleDisplay();
+  /**
+   * `createElement` - a method to produce the element DOM element for this prosemirror node.
+   * The subclasses have to implement this abstract method.
+   */
+  abstract createElement(props: {
+    node: ProsemirrorNode;
+    view: EditorView;
+    getPos: () => number;
+    options?: OptionShape;
+  }): HTMLElement;
 
-        document.removeEventListener('mousemove', onMouseMove);
-        document.removeEventListener('mouseup', onMouseUp);
+  createWrapper(
+    node: ProsemirrorNode,
+    initialSize?: { width: number; height: number },
+  ): HTMLElement {
+    const outer = document.createElement('div');
+    outer.classList.add('remirror-resizable-view');
+    outer.style.position = 'relative';
 
-        const transaction = view.state.tr.setNodeMarkup(getPos(), undefined, {
-          src: node.attrs.src,
-          width: outer.style.width,
-        });
+    if (initialSize) {
+      outer.style.width = `${initialSize.width}px`;
+      outer.style.height = `${initialSize.height}px`;
+    } else {
+      outer.style.width = node.attrs.width;
+      outer.style.height = node.attrs.height;
+    }
 
-        this.width = outer.style.width;
+    outer.style.maxWidth = '100%';
+    outer.style.minWidth = '50px';
+    outer.style.display = 'inline-block';
+    outer.style.lineHeight = '0'; // necessary so the bottom right handle is aligned nicely
+    outer.style.transition = 'width 0.15s ease-out, height 0.15s ease-out'; // make sure transition time is larger then mousemove event's throttle time
```

Comment:
We seem to doing `x.style.y` an awful lot, Could we add a helper to core-utils that does something like

```ts
setStyle(outer, {
  maxWidth: '100%',
  minWidth: '50px',
  display: 'inline-block',
  lineHeight: '0',
  transition: 'width 0.15s ease-out, height 0.15s ease-out'
})
```

Reply:
Good tips! I think we could use `Object.assign()` here.

Reply:
Looks like we added `Object.assign` everywhere, except here? 

P.s. I still think we should add a dedicated `setStyle` helper (that internally uses `Object.assign`), rather than using `Object.assign` directly.